test/e2e/framework/util.go

```diff
@@ -1856,13 +1859,14 @@ type podProxyResponseChecker struct {
 	pods           *v1.PodList
 }
 
-func PodProxyResponseChecker(c clientset.Interface, ns string, label labels.Selector, controllerName string, respondName bool, pods *v1.PodList) podProxyResponseChecker {
-	return podProxyResponseChecker{c, ns, label, controllerName, respondName, pods}
+// GetPodProxyResponseChecker returns a context for checking pods responses.
+func GetPodProxyResponseChecker(c clientset.Interface, ns string, label labels.Selector, controllerName string, respondName bool, pods *v1.PodList) PodProxyResponseChecker {
```

Comment:
Is prepending this method with `Get` something that golint enforces?

Reply:
Looks like this is a constructor method, what are your thoughts on `NewPodProxyResponseChecker` instead?

Reply:
This renaming seems for avoiding duplicated names between struct name (PodProxyResponseChecker which is returned with this method) and this method name.
`NewPodProxyResponseChecker` is also a good option for us I feel.

Reply:
done.